.devcontainer/dev/Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM python:3-slim
+
+# Add non-root user to the image.
+
+ARG USERNAME="catalyst"
+ARG USER_UID=1000
+ARG USER_GID=$USER_UID
+
+RUN groupadd --gid $USER_GID $USERNAME
+RUN useradd --uid $USER_UID --gid $USER_GID -m $USERNAME
+RUN apt-get update && apt-get install -y --no-install-recommends sudo \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+RUN echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME
+RUN chmod 0440 /etc/sudoers.d/$USERNAME
+
+# Image specific instructions: Ensure developer requirements are met.
+# Catalyst will be installed from source from within the running container.
+
+RUN apt-get update && apt-get install -y --no-install-recommends git curl patch \
+      make ninja-build clang lld ccache libomp-dev \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+USER $USERNAME
+
+RUN pip install --no-cache-dir cmake
```

Comment:
What tree does `pip install` update here? I expected to see either the `pip install --user .. ` or the `venv` setup above, but I don't see any. Does it update the system-wide tree due to some permission tricks? Or, maybe the source image include venv already?

Reply:
These commands are run as the specified user, and since the image disallows system installation (from the pip message `Defaulting to user installation because normal site-packages is not writeable`) it will be installed as a user package.

Reply:
OK. For some reason I thought it wouldn't switch to user tree without the explicit `--user` flag.